### PR TITLE
feat: Welcome Message for new members

### PR DESCRIPTION
**Pull request by itsdrawn**
This PR aims to introduce a welcome message for new members in the server

### CODE REVIEW

File: src/bot/handlers/welcome_message_handler.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        const general_chat_channel = client.channels.resolve(general_chat_id);
+        await general_chat_channel.send(`${member} due to your privacy settings not allowing me to direct message you, this embed was sent here!`, new Discord.MessageEmbed(message_contents)).catch(console.warn);
+  } 
+}
+
+//---------------------------------------------------------------------------------------------------------------//
+
+module.exports = {
+    welcomeMessageHandler,
+};
```
Comment (MidSpike) [inline src/bot/handlers/welcome_message_handler.js:47]: add an empty new-line at the end of the file

File: src/bot/handlers/welcome_message_handler.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+//---------------------------------------------------------------------------------------------------------------//
+
+async function welcomeMessageHandler(member) {
+  try {
+        const dm_channel = await member.createDM();
+        dm_channel.send(member, new Discord.MessageEmbed(message_contents)).catch(console.warn);
+  } catch { /*user has their dms off, send in general*/
+        const general_chat_channel = client.channels.resolve(general_chat_id);
+        await general_chat_channel.send(`${member} due to your privacy settings not allowing me to direct message you, this embed was sent here!`, new Discord.MessageEmbed(message_contents)).catch(console.warn);
```
Comment (MidSpike) [inline src/bot/handlers/welcome_message_handler.js:39]: Change this to the following:
```
await general_chat_channel.send(message_contents).catch(console.warn);
```

File: src/bot/handlers/welcome_message_handler.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+const general_chat_id = 601890659439476766
+
+//---------------------------------------------------------------------------------------------------------------//
+
+async function welcomeMessageHandler(member) {
+  try {
+        const dm_channel = await member.createDM();
+        dm_channel.send(member, new Discord.MessageEmbed(message_contents)).catch(console.warn);
```
Comment (MidSpike) [inline src/bot/handlers/welcome_message_handler.js:36]: change this to be the following:
```js
await dm_channel.send(message_contents);
```

File: src/bot/handlers/welcome_message_handler.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    description: [
+        'Welcome to Inertia Lighting, the largest lighting company at scale on ROBLOX sourcing some of today\'s best gear.',
+        'To get yourself started, view the following channels to obtain an understanding of our server.',
+        'Here for our product hub? View [**this!**](https://www.roblox.com/games/5438256564)',
+        'Require Support? View <#814197612491833354>**!**',
+        'Just looking to interact with the community? Go to <#601890659439476766>**!**',
+    ].join('\n'),
+}
+
+const general_chat_id = 601890659439476766
```
Comment (MidSpike) [inline src/bot/handlers/welcome_message_handler.js:29]: change this to be the following:
```js
const general_chat_id = '601890659439476766';
```

File: src/bot/handlers/welcome_message_handler.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+'use strict';
+
+//---------------------------------------------------------------------------------------------------------------//
+
+const { Discord } = require('../discord_client.js');
+
+//---------------------------------------------------------------------------------------------------------------//
+
+const message_contents = {
```
Comment (MidSpike) [inline src/bot/handlers/welcome_message_handler.js:13]: change this to the following:
```js
const message_contents = {
    content: `${member}`,
    embed: new Discord.MessageEmbed({
        author: {
            iconURL: `${client.user.displayAvatarURL({ dynamic: true })}`,
            name: `${client.user.username}`,
        },
        title: 'Welcome!',
        description: [
            'Welcome to Inertia Lighting, the largest lighting company at scale on ROBLOX sourcing some of today\'s best gear.',
            'To get yourself started, view the following channels to obtain an understanding of our server.',
            'Here for our product hub? View [**this!**](https://www.roblox.com/games/5438256564)',
            'Require Support? View <#814197612491833354>**!**',
            'Just looking to interact with the community? Go to <#601890659439476766>**!**',
        ].join('\n'),
    }),
};
```

File: src/bot/events/guild_member_add.js [hunk 6]
```diff
@@ -26,9 +27,12 @@ module.exports = {
         if (member.user.system) return; // don't operate on system accounts
         if (member.user.bot) return; // don't operate on bots to prevent feedback-loops
 
+        /* send the welcome message to the member */
+        await welcomeMessageHandler(member);
+
         /* handle nicknames for new members */
         await illegalNicknameHandler(member);
-
+
```
Comment (MidSpike) [inline src/bot/events/guild_member_add.js:35]: trim whitespace here

**Review by MidSpike — CHANGES_REQUESTED** [review]
please change the following

File: src/bot/handlers/welcome_message_handler.js (hunk 2 above)
Comment (MidSpike) [reply src/bot/handlers/welcome_message_handler.js:41]: this still needs to be fixed

File: src/bot/handlers/welcome_message_handler.js [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+                name: `${client.user.username}`,
+            },
+            title: 'Welcome!',
+            description: [
+                'Welcome to Inertia Lighting, the largest lighting company at scale on ROBLOX sourcing some of today\'s best gear.',
+                'To get yourself started, view the following channels to obtain an understanding of our server.',
+                'Here for our product hub? View [**this!**](https://www.roblox.com/games/5438256564)',
+                'Require Support? View <#814197612491833354>**!**',
+                'Just looking to interact with the community? Go to <#601890659439476766>**!**',
+            ].join('\n'),
```
Comment (Will3126) [inline src/bot/handlers/welcome_message_handler.js:32]: should be embeds: []

File: src/bot/handlers/welcome_message_handler.js [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+        }),
+    };
+
+    try {
+        const dm_channel = await member.createDM();
+        await dm_channel.send(message_contents);
```
Comment (Will3126) [inline src/bot/handlers/welcome_message_handler.js:38]: send({conent: message_contents})

**Review by Will3126 — CHANGES_REQUESTED** [review]
Needs to be able to work with Discord.JS v13

File: src/bot/handlers/welcome_message_handler.js (hunk 8 above)
Comment (itsdrawn) [reply src/bot/handlers/welcome_message_handler.js:38]: content*

File: src/bot/handlers/welcome_message_handler.js (hunk 7 above)
Comment (MidSpike) [reply src/bot/handlers/welcome_message_handler.js:32]: we are aiming for v12 compatibility not v13

File: src/bot/handlers/welcome_message_handler.js (hunk 8 above)
Comment (MidSpike) [reply src/bot/handlers/welcome_message_handler.js:38]: we are aiming for v12 compatibility not v13

File: src/bot/handlers/welcome_message_handler.js [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+    const message_contents = {
+        content: `${member}`,
+        embed: new Discord.MessageEmbed({
+            author: {
+                iconURL: `${client.user.displayAvatarURL({ dynamic: true })}`,
+                name: `${client.user.username}`,
+            },
+            title: 'Welcome!',
+            description: [
+                'Welcome to Inertia Lighting, the largest lighting company at scale on ROBLOX sourcing some of today\'s best gear.',
```
Comment (MidSpike) [inline src/bot/handlers/welcome_message_handler.js:27]: Change the following to:
```js
[
    'Welcome to the Inertia Lighting discord server!',
    'To get started, visit the following channels to learn about our server:',
    '- <#601890659439476766>', // general channel
    '- <#814197612491833354>', // support channel
    'Also, check out our [product hub](https://www.roblox.com/games/5438256564) to purchase our products!',
].join('\n'),
```

**Review by MidSpike — CHANGES_REQUESTED** [review]

File: src/bot/handlers/welcome_message_handler.js (hunk 9 above)
Comment (MidSpike) [reply src/bot/handlers/welcome_message_handler.js:27]: I also recommend adding more channels

**Review by Will3126 — APPROVED** [review]
Looks good now!

**Review by Will3126 — CHANGES_REQUESTED** [review]
Create a variable for each channel that is being manually referenced.

Example:
https://github.com/Inertia-Lighting/discord-bot-api-server/pull/72/files#diff-ee7b620454781575d7b8546b9d5a7d17fbbb42d02a20ea56b20465d42eaae36cR13

Change these to use variables
https://github.com/Inertia-Lighting/discord-bot-api-server/pull/72/files#diff-ee7b620454781575d7b8546b9d5a7d17fbbb42d02a20ea56b20465d42eaae36cR29-R33

File: src/bot/handlers/welcome_message_handler.js [hunk 10]
```diff
@@ -10,7 +10,11 @@ const { Discord } = require('../discord_client.js');
 
 //---------------------------------------------------------------------------------------------------------------//
 
-const general_chat_id = '601890659439476766';
+const general_chat_id = '<#601890659439476766>';
+const support_channel_id = '<#814197612491833354>';
+const rules_channel_id = '<#601890376667889684>';
+const info_channel_id = '<#737156807910359091>';
+const news_channel_id = '<#854442081899642950>';
```
Comment (Will3126) [inline src/bot/handlers/welcome_message_handler.js:17]: Make them just the channel ids, not the channel mentions

File: src/bot/handlers/welcome_message_handler.js [hunk 11]
```diff
@@ -26,11 +30,11 @@ async function welcomeMessageHandler(member) {
...
-                '- <#601890659439476766>', // general channel
-                '- <#814197612491833354>', // support channel
-                '- <#601890376667889684>', // rules
-                '- <#737156807910359091>', // info
-                '- <#854442081899642950>', // news
+                `- ${general_chat_id}`, // general channel id
+                `- ${support_channel_id}`, // support channel id
+                `- ${rules_channel_id}`, // rules channel id
+                `- ${info_channel_id}`, // info channel id
+                `- ${news_channel_id}`, // news channel id
```
Comment (Will3126) [inline src/bot/handlers/welcome_message_handler.js:37]: add the channel mentions here.

**Review by Will3126 — CHANGES_REQUESTED** [review]

**Review by MidSpike — APPROVED** [review]
Things look good now, good job drawn.

**Review by Will3126 — APPROVED** [review]
All looks good!

**Review by rossloika — APPROVED** [review]

**Pull request merged**